### DISCUSSION TITLE
Оставить в плейлисте треки только незнакомых исполнителей

### DISCUSSION THREAD

**Discussion by F000x** [Chimildic/YaMuTools#60] [Идея]
Очень хотелось бы иметь возможность оставить в плейлисте треки только незнакомых исполнителей, а треки, в которых участвуют исполнители, которых ты уже слушал, удалить. Исполнителей можно брать из истории или с lastfm, если так проще

**Comment by kisoid** [comment]
Или можно реализовать атомарную операцию "Вычитание исполнителей": https://github.com/Chimildic/YaMuTools/discussions/56
и на основе неё сделать любую комплекную, включая эту, путём вычитания тех исполнителей, которые знакомы.

**Comment by Chimildic** [comment]
Почему не подходит функция _фильтр треков_ > _удалить недавно игравшие_?

**Reply by kisoid** [reply]
Отвечу за топикстартера: потому что в плейлисте могут находиться _не_знакомые треки (т.е. отсутствующие в истории прослушивания) знакомых исполнителей, которые юзер тоже хотел бы удалить (т.к. он хочет удалить "знакомых" исполнителей).
Другой вопрос: а как скрипт должен понять, какой исполнитель знакомый, а какой нет?
Сколько треков исполнителя должно быть прослушано? 10? 100? 1000?
Наиболее общий ответ: никак. Потому что у всех своё субъективное понимание значения "знакомое". Кто-то может годами слушать исполнителя фоном, неосознанно. Это не означает, что данный исполнитель знаком, хоть и присутствует в истории.
Поэтому оптимальнее реализовать функцию "Вычитание исполнителей" https://github.com/Chimildic/YaMuTools/discussions/56 и пусть юзер явно укажет, кого хочет удалить из плейлиста.
